src/ApparentHorizons/Strahlkorper.hpp

```diff
@@ -0,0 +1,119 @@
+  /// as the `coefficients` member function returns.
+  Strahlkorper(const Strahlkorper& another_strahlkorper,
+               DataVector coefs) noexcept;
+  /// Construct a Strahlkorper from a DataVector containing the radius
+  /// at the collocation points.
+  Strahlkorper(const DataVector& radius_at_collocation_points, size_t l_max,
+               size_t m_max, std::array<double, 3> center) noexcept;
+
+  /// Serialization for Charm++
+  void pup(PUP::er& p);  // NOLINT
+
+  /*!
+   *  These coefficients are stored as SPHEREPACK coefficients.
+   *  Suppose you represent a set of coefficients \f$F^{lm}\f$ in the expansion
+   *  \f[
+   *  f(\theta,\phi) =
+   *  \sum_{l=0}^{l_max} \sum_{m=-l}^{l} F^{lm} Y^{lm}(\theta,\phi)
+   *  \f]
+   *  Here the \f$Y^{lm}(\theta,\phi)\f$ are the usual complex-valued scalar
+   *  spherical harmonics, so \f$F^{lm}\f$ are also complex-valued.
+   *  But here we assume that \f$f(\theta,\phi)\f$ is real, so therefore
+   *  the \f$F^{lm}\f$ obey \f$F^{l-m} = (-1)^m (F^{lm})^\star\f$, so one
+   *  does not need to store both real and imaginary parts.
+   *  So:
+   *  strahlkorper_coefs_(l,m) = \f$(-1)^m Q Re(F^{lm})\f$ for \f$m\ge 0\f$
+   *  strahlkorper_coefs_(l,m) = \f$(-1)^m Q Im(F^{lm})\f$ for \f$m<   0\f$
+   *  Where Q = sqrt(2.0/M_PI)
+   */
+  SPECTRE_ALWAYS_INLINE const DataVector& coefficients() const noexcept {
+    return strahlkorper_coefs_;
+  }
+
+  /// Point about which the spectral basis of the Strahlkorper is expanded.
+  /// The center is given in the frame in which the Strahlkorper is defined.
+  /// This center must be somewhere inside the Strahlkorper, but in principle
+  /// it can be anywhere.  See physical_center for a different measure.
+  SPECTRE_ALWAYS_INLINE const std::array<double, 3>& center() const noexcept {
+    return center_;
+  }
+
+  /// Approximate physical center (determined by l=1 coefficients)
+  /// Implementation of Eqs. (38)-(40) in Hemberger et al, arXiv:1211.6079
+  std::array<double, 3> physical_center() const noexcept;
+
+  /// Average radius of the surface (determined by l=0 Ylm coefficient)
+  double average_radius() const noexcept;
+
+  /// Maximum \f$l\f$ in \f$Y_{lm}\f$ decomposition.
+  SPECTRE_ALWAYS_INLINE size_t l_max() const noexcept { return l_max_; }
+
+  /// Maximum \f$m\f$ in \f$Y_{lm}\f$ decomposition.
+  SPECTRE_ALWAYS_INLINE size_t m_max() const noexcept { return m_max_; }
+
+  /// Radius at a particular angle \f$(\theta,\phi)\f$.
+  /// This is inefficient if done at multiple points many times.
+  /// See YlmSpherepack for alternative ways of computing this.
+  double radius(double theta, double phi) const noexcept;
+
+  /// Determine if a point `x` is contained inside the surface.
+  /// The point must be given in Cartesian coordinates in the frame in
+  /// which the Strahlkorper is defined and relative to the origin `center`.
+  /// This is inefficient if done at multiple points many times.
```

Comment:
Should there be a `See YlmSpherepack` here too, or is there no more efficient way?

Reply:
There isn't a more efficient way implemented right now.
But there could be a `points_are_contained()` function that takes a vector of points and returns a vector of bools, which would be more efficient because it could interpolate to all the points at once.  Not sure if I should add such a function now (which may not be needed) or wait until it is needed.